### PR TITLE
refactor(lodash): replace orderBy

### PR DESCRIPTION
**Pull request by Haroenv**
I generated this code:

1. take original lodash code
2. inline imports
3. inline helpers
4. remove edge-case helpers (string instead of array arguments, path instead of element, Symbol support, NaN support etc.)
5. copy tests from lodash
6. change tests to work in Jest
7. remove one test that no longer would have passed

This probably can be simplified a bit more here and there, but likely isn't that important

### CODE REVIEW

File: src/functions/orderBy.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      value: value
+    };
+  });
+
+  result.sort(function comparer(object, other) {
+    var index = -1;
+
+    while (++index < object.criteria.length) {
+      var res = compareAscending(object.criteria[index], other.criteria[index]);
+      if (res) {
```
Comment (eunjae-lee) [inline src/functions/orderBy.js]: We could change this block to 

```js
if (res) {
  if (orders[index] === 'desc') {
    return -res;
  } else {
    return res;
  }
}
```

If `index` is out of range, it will fail `=== 'desc'` anyway.

Comment (Haroenv) [reply src/functions/orderBy.js]: I was thinking about that, but I think this was added in lodash to shortcut the loop, although reading it again, it might not make a difference

Comment (eunjae-lee) [reply src/functions/orderBy.js]: hmmm no idea 🤷🏼‍♂️

File: src/functions/orderBy.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+/**
+ * @param {Array<object>} collection object with keys in attributes
+ * @param {Array<string>} iteratees attributes
+ * @param {Array<string>} orders asc | desc
+ */
+function orderBy(collection, iteratees, orders) {
+  if (!Array.isArray(collection)) {
+    return [];
```
Comment (eunjae-lee) [inline src/functions/orderBy.js]: We can add a test case for this.

File: src/functions/orderBy.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  }
+
+  if (!Array.isArray(orders)) {
+    orders = [];
```
Comment (eunjae-lee) [inline src/functions/orderBy.js]: We can add a test case for this.

Comment (Haroenv) [reply src/functions/orderBy.js]: this has a test case: https://github.com/algolia/algoliasearch-helper-js/pull/698/files#diff-67016e5575af72c2e6fbd33e554c7988R31

File: src/functions/orderBy.js (hunk 2 above)
Comment (Haroenv) [reply src/functions/orderBy.js]: cc320ed3de6798a58e82a151e33f5092cd486112
